### PR TITLE
chore: manage dogfood template using coderd provider

### PR DESCRIPTION
**Pull request by ethanndickson**

### CODE REVIEW

**Comment by ethanndickson** [comment]
* **#14321** <a href="https://app.graphite.dev/github/pr/coder/coder/14321?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ethanndickson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/dogfood.yaml [hunk 1]
```diff
@@ -101,22 +104,16 @@ jobs:
...
         env:
           # Consumed by Coder CLI
           CODER_URL: https://dev.coder.com
           CODER_SESSION_TOKEN: ${{ secrets.CODER_SESSION_TOKEN }}
           # Template source & details
-          CODER_TEMPLATE_NAME: ${{ secrets.CODER_TEMPLATE_NAME }}
-          CODER_TEMPLATE_VERSION: ${{ steps.vars.outputs.sha_short }}
-          CODER_TEMPLATE_DIR: ./dogfood
-          CODER_TEMPLATE_MESSAGE: ${{ steps.message.outputs.pr_title }}
+          TF_VAR_coder_template_name: ${{ secrets.CODER_TEMPLATE_NAME }}
```
Comment (matifali) [inline .github/workflows/dogfood.yaml:116]: Is it case insensitive?

Comment (ethanndickson) [reply .github/workflows/dogfood.yaml:116]: Apparently so

Comment (ethanndickson) [reply .github/workflows/dogfood.yaml:116]: Yeah, I can't get it to pick up the all-caps version:
```tf
variable "CODER_TEMPLATE_NAME" {
  type = string
}
```
```bash
$ export CODER_TEMPLATE_NAME="coder"
$ echo "$CODER_TEMPLATE_NAME"
coder
```

```
│ Error: Invalid Attribute Value Length
│ 
│   with coderd_template.dogfood,
│   on main.tf line 45, in resource "coderd_template" "dogfood":
│   45:   name            = var.CODER_TEMPLATE_NAME
│ 
│ Attribute versions[0].name string length must be at least 1, got: 0
```

Comment (matifali) [reply .github/workflows/dogfood.yaml:116]: Shouldn't it be
```
$ export TF_VAR_CODER_TEMPLATE_NAME="coder"
```
For terraform to see it as an input variable?

Comment (matifali) [reply .github/workflows/dogfood.yaml:116]: So you are doing it the correct way. :)

Comment (ethanndickson) [reply .github/workflows/dogfood.yaml:116]: Yeah, oops, I typo'd it the second time around. I must've also done something wrong the first time - fixed, thank you :)

**Review by matifali — APPROVED** [review]
Looks good to me. I will suggest testing once with a `terraform apply`. 
Also make sure the secrets are set in GitHub actions.

**Review by deansheather — APPROVED** [review]

**Pull request merged**